src/main/kotlin/calculator/parser/SimpleNodeParser.kt

```diff
@@ -0,0 +1,45 @@
+package calculator.parser
+
+import calculator.calculator.Node
+import calculator.calculator.OperatorString
+import java.util.regex.Pattern
+
+class SimpleNodeParser : NodeParser {
+    override fun parse(text: String): List<Node> {
+        return text
+            .split(Regex(REGEX_FOR_NODE))
+            .trimAll()
+            .exceptEmpty()
+            .convertAllToNode()
+    }
+
+    fun List<String>.trimAll() = map { it.trim() }
+
+    fun List<String>.exceptEmpty() = filter { it.isNotEmpty() }
+
+    fun String.convertToNode(): Node {
+        takeIf {
+            Pattern.compile(PATTERN_FOR_NUMBER)
+                .matcher(it).matches()
+        }?.let {
+            return Node.Number(it.toDouble())
```

Comment:
생성자의 수가 늘어날수록 클래스를 사용하기 위한 유연성은 높아집니다.

```suggestion
            return Node.Number(it)
```